src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1362,29 +1157,31 @@ private void updateToolFlag(@NotNull final String tool, final int required)
      */
     private int getMostEfficientTool(@NotNull final Block target)
     {
-        final String tool;
+        final IToolType toolType;
         final int required;
+        //TODO this should not be here
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)